src/snap_management.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+# Learn more at: https://juju.is/docs/sdk
+
+"""Snap Installation Module.
+
+Modified from https://github.com/canonical/k8s-operator/blob/main/charms/worker/k8s/src/snap.py
+"""
+
+
+import logging
+import platform
+
+import charms.operator_libs_linux.v2.snap as snap_lib
+from charms.operator_libs_linux.v2.snap import SnapState
+
+# Log messages can be retrieved using juju debug-log
+log = logging.getLogger(__name__)
+
+
+# Map of the grafana-agent snap revision to install for given architectures and strict mode.
+_grafana_agent_snap_spec = {
+    "strict": {
+        "amd64": {
+            "name": "grafana-agent",
+            "revision": "16",  # 0.35.0
+        },
+        "arm64": {
+            "name": "grafana-agent",
+            "revision": "23",  # 0.39.2
+        },
+    },
+}
+
+
+def install_ga_snap(classic: bool = False):
+    """Looks up system details and installs the appropriate grafana-agent snap revision."""
+    arch = get_system_arch()
+    confinement = "classic" if classic else "strict"
+    snap_spec = _grafana_agent_snap_spec[confinement][arch]
+    _install_snap(name=snap_spec["name"], revision=snap_spec["revision"], classic=classic)
+
+
+def _install_snap(
+    name: str,
+    revision: str,
+    classic: bool = False,
+):
+    """Install the given snap revision, holding it so it won't update."""
+    cache = snap_lib.SnapCache()
+    snap = cache[name]
+    log.info(
+        f"Ensuring {name} snap is installed at revision={revision}"
+        f" with classic confinement={classic}"
+    )
+    snap.ensure(state=SnapState.Present, revision=revision, classic=classic)
+    snap.hold()
+
+
+def get_system_arch() -> str:
+    """Returns the architecture of this machine, mapping some values to amd64 or arm64.
+
+    If platform is x86_64 or amd64, it returns amd64.
+    If platform is aarch64, arm64, armv8b, or armv8l, it returns arm64.
+    """
+    arch = platform.processor()
+    if arch in ["x86_64", "amd64"]:
+        arch = "amd64"
+    elif arch in ["aarch64", "arm64", "armv8b", "armv8l"]:
```

Comment:
why isn't this an else? 
just for my peace of mind, should we add an else and raise some error?

Reply:
This comes from [here](https://github.com/canonical/loki-k8s-operator/blob/main/lib/charms/loki_k8s/v1/loki_push_api.py#L1853-L1859).  We're converting known common aliases to a single name, but unknown names are still allowed.  We don't need an `else` for the code itself, but I'll add a comment to clarify as it does seem like an oversight.